internal/otelcollector/config/metric/agent/receivers_test.go

```diff
@@ -30,39 +30,74 @@ func TestReceivers(t *testing.T) {
 
 	t.Run("runtime input enabled verify k8sClusterReceiver", func(t *testing.T) {
 		agentNamespace := "test-namespace"
-		collectorConfig := sut.Build([]telemetryv1alpha1.MetricPipeline{
-			testutils.NewMetricPipelineBuilder().WithRuntimeInput(true).Build(),
-		}, BuildOptions{
-			AgentNamespace: agentNamespace,
-		})
 
-		require.Nil(t, collectorConfig.Receivers.PrometheusAppPods)
-		require.Nil(t, collectorConfig.Receivers.PrometheusIstio)
+		tests := []struct {
+			name                  string
+			pipeline              telemetryv1alpha1.MetricPipeline
+			expectedMetricsToDrop K8sClusterMetricsToDrop
+		}{
+			{
+				name: "default resources enabled",
```

Comment:
when testing for default resources, pod and container metrics should not be enabled explicitly